### PR TITLE
Bump node to version 14

### PR DESCRIPTION
**Pull request by lynzt**
# Description

Update Node from 12 to 14
This update based on this node bump [this PR](https://github.com/transcom/circleci-docker/pull/37)

## Changelog or Releases

- [node 14.X](https://nodejs.org/en/about/releases/)

## Questions

To the question about version @monfresh posed below, how is https://github.com/CircleCI-Public/circleci-dockerfiles/tree/master/node/images "created" and how does it determine what images are there? Do we have control over this?

I'm also still a bit confused as to what/how the images are generated and pushed to dockerhub - particularly, what are `milmove-app-browsers`, `milmove-atlantis`, `milmove-infra-tf*` (and things like `milmove-cypress` and `milmove-cypress-master`

### CODE REVIEW

File: .node-version [hunk 1]
```diff
@@ -1 +1 @@
-12.16.3
+14.16.1
```
Comment (monfresh) [inline .node-version:1]: Why not 14.17.1, which is the latest LTS version?

Comment (lynzt) [reply .node-version:1]: I was basing the version off of what I saw here: https://github.com/CircleCI-Public/circleci-dockerfiles/tree/master/node/images

if this is not correct, i'm happy to bump it up, but I was using the latest 14x that was listed in the above directory

**Review by travelar — APPROVED** [review]

**Review by mdawn — APPROVED** [review]
> I'm also still a bit confused as to what/how the images are generated and pushed to dockerhub - particularly, what are...

I'm not quite sure what you mean by how the images are generated. I **_can_** offer that:
- the `tf*` images are used by infra to bump up terraform versions
- the `milmove-atlantis` image is used in running [Atlantis](https://www.runatlantis.io/) in `exp` (see [here for the module call](https://github.com/transcom/transcom-infrasec-gov/blob/f75e8173f9e5c9c8db7ab39ce813055a490ff537/transcom-gov-milmove-exp/atlantis-global/main.tf#L15))

Hope that helps!

File: .node-version (hunk 1 above)
Comment (monfresh) [reply .node-version:1]: I don't think the version of node comes from the CircleCI images. We should be able to install whatever version we want. I think that comment in the Dockerfile is confusing. I'm pretty sure the only reason that Circle CI URL is mentioned is to grab the command to install chrome. That's all there is to that. If you look in `milmove-app/Dockerfile `, you will see we get node from `deb https://deb.nodesource.com/node_14.x $(lsb_release -sc) main`, and that has the latest 14.17.1.

**Review by monfresh — APPROVED** [review]
Looks great! Thanks for updating to 14.17.1.

**Review by reggieriser — APPROVED** [review]
LGTM, with one aside...

For future reference, just want to capture what @lynzt and I discussed in a Zoom call this past Friday.  Most of it can be found in [this Slack thread](https://ustcdp3.slack.com/archives/CP6PTUPQF/p1624659946482800?thread_ts=1624655740.481700&cid=CP6PTUPQF).  In a nutshell, we don't appear to pin the node version to a specific minor/patch version level here (nor did we before with 12.x), so anytime this image gets rebuilt (like via a Go upgrade) and referenced from milmove, the node version may be moving forward as well to a later minor/patch version without us realizing it.  So, on the milmove side, we tend to stick to the version (like `14.17.1`) that was in effect when this PR went in and the image was created rather than using the actual version that's in the image in circle and the deployed environments.

**Comment by monfresh** [comment]
Agreed. I think we should write a JIRA story for this if one doesn't exist and tag it as a TRA item (by selecting the Component "Technology Updates / Fixes")

**Pull request merged**